keras_cv/models/__internal__/darknet_utils.py

```diff
@@ -0,0 +1,398 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DarkNet model utils for KerasCV.
+Reference:
+  - [YoloV3 Paper](https://arxiv.org/abs/1804.02767)
+  - [YoloV3 implementation](https://github.com/ultralytics/yolov3)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class DarknetConvBlock(layers.Layer):
+    """The basic conv block used in Darknet. Applies Conv2D followed by a BatchNorm.
+
+    Args:
+        filters: Integer, the dimensionality of the output space (i.e. the number of
+            output filters in the convolution).
+        kernel_size: An integer or tuple/list of 2 integers, specifying the height
+            and width of the 2D convolution window. Can be a single integer to specify
+            the same value both dimensions.
+        strides: An integer or tuple/list of 2 integers, specifying the strides of
+            the convolution along the height and width. Can be a single integer to
+            the same value both dimensions.
+        use_bias: Boolean, whether the layer uses a bias vector.
+        activation: the activation applied after the BatchNorm layer. One of "silu",
+            "relu" or "leaky_relu". Defaults to "silu".
+    """
+
+    def __init__(
+        self, filters, kernel_size, strides, use_bias=False, activation="silu", **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.filters = filters
+        self.kernel_size = kernel_size
+        self.strides = strides
+        self.use_bias = use_bias
+        self.activation = activation
+
+        self.conv = layers.Conv2D(
+            filters,
+            kernel_size,
+            strides,
+            padding="same",
+            use_bias=use_bias,
+        )
+
+        self.batch_norm = layers.BatchNormalization()
+
+        if activation == "silu":
+            self.act = layers.Lambda(lambda x: keras.activations.swish(x))
+        elif activation == "relu":
+            self.act = layers.ReLU()
+        elif activation == "leaky_relu":
+            self.act = layers.LeakyReLU(0.1)
+
+    def call(self, x):
```

Comment:
I think we can replace this entire block with a sequential model:

def DarknetConvBlock(layers.Layer):
    layers = [
        createConvAsYouDidAbove,  BatchNorm, activationAsYouDidAbove
    ]
    return keras.Sequential(layers)

Reply:
Can this cause serialization issues? That being said, I think it will definitely be more concise.

Reply:
no, Sequential models are layers, and models are also layers.  so serialization wise this is fine.

Reply:
done!